articles/client-platforms/angular2/04-linking-accounts.md

```diff
@@ -0,0 +1,206 @@
+    allowedConnections: ['Username-Password-Authentication', 'facebook', 'google-oauth2'],
+    languageDictionary: { // allows to override dictionary entries
+      title: "Link with:"
+    }
+  });
+```
+
+Then, when setting the `authenticated` callbacks we can know which login is, checking the `authResult.state` attribute. 
+
+```typescript
+...
+// Add callback for lock `authenticated` event
+this.lock.on("authenticated", (authResult) => {
+  // Every lock instance listen to the same event, so we have to check if
+  // it's not the linking login here.
+  if(authResult.state != "linking"){
+    localStorage.setItem('id_token', authResult.idToken);
+    this.fetchProfile(authResult.idToken);
+  }
+});
+
+// Add callback for lockLink `authenticated` event
+this.lockLink.on("authenticated", (authResult) => {
+  // Every lock instance listen to the same event, so we have to check if
+  // it's the linking login here.
+  if(authResult.state == "linking"){
+    // If it's the linking login, then do the link through the API.
+    this.doLinkAccounts(authResult.idToken);
+  }
+});
+...
+```
+
+Now that we already have the second login handled let's see how to actually do the linking.
+To call the api, `angular-jwt` provides a `AuthHttp` class which automatically attach the user's JWT as an `Authorization` header when making requests.
+
+First you need to add the AUTH_PROVIDERS from angular-gwt
+
+${snippet(meta.snippets.auth-providers-setup)}
+
+Then you can import AuthHttp in your component and make the authenticated request:
+
+
+```typescript
+public doLinkAccounts(accountToLinkJWT) {
+  var headers: any = {
+    'Accept': 'application/json',
+    'Content-Type': 'application/json'
+  };
+
+  var data: any = JSON.stringify({
+    link_with: accountToLinkJWT
+  });
+
+  this.authHttp
+    .post('https://' + '${account.namespace}' + '/api/v2/users/' + this.userProfile.user_id + '/identities', data, {headers: headers})
+    .map(response => response.json())
+    .subscribe(
+      response => {
+        console.log("accounts linked");
+        this.fetchProfile(localStorage.getItem('id_token'));
+        this.router.navigate(['Profile']);
+      },
+      error => alert(error.json().message)
+    );
+}
+```
+
+The method takes the token of the account to link with, so we just make a post to the api, passing in the body the parameter `link_with` parameter with the JWT value.
+We fetched the profile on success and we can see that the accounts are now linked.
+
+
+To make everything works just call `show` method on `lockLink` instance:
+
+```typescript
+public linkAccount() {
+  this.lockLink.show();
+}
+```
+
+## User profile linked accounts information
+
+User's profile contains an array of identities which is made of profile information from other providers. You can see this by accessing the `Users` page from the dashboard, select a user and scroll down to the identities. This is what it looks like after linking Facebook:
```

Comment:
I think it's nice to make clear it's auth0 website dashboard, perhaps adding a link